files/common/usr/bin/delphix-startup-screen

```diff
@@ -0,0 +1,330 @@
+import time
+import subprocess
+
+STATUS_HEADER = "STATUS\t\t\tSERVICE\t\t\t"
+LAYOUT_STR = "Keyboard layout (press 9 to change):"
+CONSOLE_STR = "Press any other key for console access."
+KEY_LAYOUT_SELECT_STR = "Select a keyboard layout by number [1-%d]:"
+
+log_file = "/tmp/dpx_startup_screen.log"
+logo = "Delphix Engine Version "
+
+website = "www.delphix.com"
+
+WIN_LEN = 0
+WIN_HT = 0
+
+def get_svcs():
+    cmd = ['systemctl', 'show', '-p',  'Wants', 'delphix.target']
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    return cp.stdout.split("=", 1)[1].split()
+
+def run_svcs(options=None):
+    '''
+    Run the command to retrieve the status of all the services. The options
+    argument can be used to obtain additional output from the 'systemctl'
+    command. By default we only obtain the 'ActiveState' of the services.
+    '''
+
+    fields = 'ActiveState'
+    cmd = ['systemctl', 'show', '--value', '-p']
+    if options:
+        fields += ',' + options
+    cmd.append(fields)
+    for svc in get_svcs():
+        cmd.append(svc)
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    return cp.stdout
+
+def get_keyboard_layout():
+    cmd = ['localectl', 'status']
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    output = cp.stdout.split()
+    return output[output.index("Layout:") + 1]
+
+def set_keyboard_layout(layout):
+    cmd = ['localectl', 'set-x11-keymap', layout, 'pc105']
+    cp = subprocess.run(cmd, shell=False)
+    return subprocess.run('setupcon', shell=False)
+
+def get_valid_keyboard_layouts():
+    cmd = ['localectl', '--no-pager', 'list-x11-keymap-layouts']
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    return cp.stdout.splitlines()
+
+def load_header(stdscr):
+    global logo
+
+    cmd = ['/opt/delphix/server/bin/get_appliance_version']
+    cp = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True)
+    version = cp.stdout.split('-', 1)[0]
+
+    stdscr.clear()
+    stdscr.addstr(1, 2, logo + str(version), curses.A_BOLD)
+    stdscr.addstr(1, WIN_LEN - len(website) - 2, website,
+                  curses.A_BOLD)
+
+    win = stdscr.subwin(WIN_HT - 4, WIN_LEN - 4, 3, 2)
+    win.bkgd(' ', curses.color_pair(2))
+    win.box()
+    return win
+
+def is_server_down():
+    '''
+    Determine if the server is down by looking at the state of all the
+    services. If any service is 'failed' then we declare the server down.
+    '''
+    out = run_svcs()
+    for i in out.splitlines(False):
+        #
+        # Skip any empty lines
+        #
+        if not i.strip():
+            continue
+
+        if i.strip() == "failed":
+            return True
+    return False
+
+def is_server_up():
+    '''
+    Determine if all the services are 'active' or 'inactive'. Inactive
+    is considered up since the service may not be enabled.
+    '''
+    out = run_svcs().rstrip("\n")
```

Comment:
Again what is different here from `is_server_down()` that we have the extra `rstrip("\n")`?

Reply:
This is not necessary and I've removed this.